pkg/feature/flagtrigger/webhook/handler.go

```diff
@@ -0,0 +1,284 @@
+// Copyright 2023 The Bucketeer Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package webhook
+
+import (
+	"context"
+	"encoding/base64"
+	"net/http"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/bucketeer-io/bucketeer/pkg/crypto"
+	featureclient "github.com/bucketeer-io/bucketeer/pkg/feature/client"
+	"github.com/bucketeer-io/bucketeer/pkg/feature/domain"
+	v2fs "github.com/bucketeer-io/bucketeer/pkg/feature/storage/v2"
+	"github.com/bucketeer-io/bucketeer/pkg/log"
+	"github.com/bucketeer-io/bucketeer/pkg/storage/v2/mysql"
+	featureproto "github.com/bucketeer-io/bucketeer/proto/feature"
+)
+
+type handler struct {
+	mysqlClient       mysql.Client
+	featureClient     featureclient.Client
+	triggerCryptoUtil crypto.EncrypterDecrypter
+	logger            *zap.Logger
+}
+
+type Option func(*options)
+
+type options struct {
+	logger *zap.Logger
+}
+
+func WithLogger(l *zap.Logger) Option {
+	return func(opts *options) {
+		opts.logger = l
+	}
+}
+
+func NewHandler(
+	mysqlClient mysql.Client,
+	featureClient featureclient.Client,
+	triggerCryptoUtil crypto.EncrypterDecrypter,
+	opts ...Option,
+) *handler {
+	options := &options{
+		logger: zap.NewNop(),
+	}
+	for _, o := range opts {
+		o(options)
+	}
+	return &handler{
+		mysqlClient:       mysqlClient,
+		featureClient:     featureClient,
+		triggerCryptoUtil: triggerCryptoUtil,
+		logger:            options.logger,
+	}
+}
+
+func (h handler) ServeHTTP(resp http.ResponseWriter, request *http.Request) {
```

Comment:
We don't need an HTTP server.
The envoy transcodes the JSON request to GRPC automatically.

This API is transcoded by envoy.
https://github.com/bucketeer-io/bucketeer/blob/main/proto/gateway/service.proto#L113-L117

Reply:
Thanks for suggestion! 
I'll convert this webhook into GRPC API.